src/charm.py

```diff
@@ -72,9 +72,9 @@ def __init__(self, *args):
             self,
             relation_name=PEER_RELATION_NAME,
             additional_secret_fields=[
-                AUTH_FILE_DATABAG_KEY,
-                CFG_FILE_DATABAG_KEY,
-                MONITORING_PASSWORD_KEY,
+                self._translate_field_to_secret_key(AUTH_FILE_DATABAG_KEY),
+                self._translate_field_to_secret_key(CFG_FILE_DATABAG_KEY),
+                self._translate_field_to_secret_key(MONITORING_PASSWORD_KEY),
```

Comment:
Needed the transformed peer data keys to properly register as secrets.

Reply:
Yes, but not just here...

`get/set_secret()` functions need to have it translated too, when passing the secret key to the lib.